link-grammar/linkage/linkage.c

```diff
@@ -875,6 +875,38 @@ const char * linkage_get_word(const Linkage linkage, WordIdx w)
 	return linkage->word[w];
 }
 
+// Need to free the returned value of this function later:
+// the memory pointed to by bare_word
+char * linkage_get_bare_word(const Linkage linkage, WordIdx w)
+{
+	if (!linkage) return NULL;
+	if (linkage->num_words <= w) return NULL; /* bounds-check */
+
+	// first look for unknown word tag
+	char delimiter = '[';
+	char *separator_ptr = strrchr(linkage->word[w], delimiter);
+
+	if(separator_ptr == NULL)
+	{
+	    // look for beginning of word tag
+		delimiter = '.';
+	    separator_ptr = strrchr(linkage->word[w], delimiter);
+	    if(separator_ptr == NULL)
+		{
+			// if no tags are aded, point to the end of the word
+			delimiter = '\0';
+	  		separator_ptr = strrchr(linkage->word[w], delimiter); // if there is no separator in the word
+		}
+	}
+	// copy and return word without the suffix
+	int position = separator_ptr - linkage->word[w];
+	char* bare_word = (char*) malloc((position + 1) * sizeof(char));
+	memcpy(bare_word, linkage->word[w], position);
```

Comment:
I just realized that 
    linkage->wg_path_display[word_index]->subword
contains the indexed bare word, without extra tags, so that's probably a way to get them directly, unless I'm missing some possible suffix I have not found so far.